### ISSUE TITLE
508-defect-2 [SCREENREADER, COGNITION]: if links are different, they MUST communicate their differences

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#12414]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Screen reader users may navigate a page by keyboard, moving from one focusable area to another. If links have the same label but the interior content will be unique for the individual use case, there **must** be text to distinguish the links and their unique purpose. 
- At a minimum, this would be provided in screen reader only text, since they lack the visual cues to associate the link with the surrounding content and may only be hearing links. 
- Alternatively, the links could have an `aria-label` to distinguish the links. 
- It may be most helpful for all if the link text is clear for all, as those with cognitive conditions may struggle to understand the contents will be different just from placement alone. 

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to navigate a web page by keyboard and understand the differences between interactive areas.

## Environment

* Operating System: all
* Browser: all
* Screenreader: all
* Server destination: staging

## Steps to Recreate

1. Logged in as user 1, enter https://staging.va.gov/manage-va-debt/your-debt/?postLogin=true#content in browser
2. Start screenreading device listed in Environment
3. Tab through the page content
4. Verify that the Why might I have this debt? link is repeated, without the surrounding context communicated for non-sighted users.

## Possible Fixes (optional)

Three possible fixes:
1. Add screen reader only text to communicate the differences in links
1. Use a unique `aria-label` for each link to distinguish between the links
1. Change the link text to "Why might I have post 9/11 GI Bill debt for housing?"

### Current code

```diff
<button type="button" class="additional-info-button va-button-link vads-u-display--block" aria-expanded="true" aria-controls="tooltip-3">
  <span class="additional-info-title">
    Why might I have this debt?
    <i class="fas fa-angle-down open"></i>
  </span>
</button>
```

### Potential fix

```diff
<button type="button" class="additional-info-button va-button-link vads-u-display--block" aria-expanded="true" aria-controls="tooltip-3">
  <span class="additional-info-title">
+    Why might I have this <span class="sr-only">post 9/11 GI Bill debt for housing</span> debt?
    <i class="fas fa-angle-down open"></i>
  </span>
</button>
```

### ⭐  Recommended fix

```diff
<button type="button" class="additional-info-button va-button-link vads-u-display--block" aria-expanded="true" aria-controls="tooltip-3">
  <span class="additional-info-title">
+    Why might I have this Post 9/11 GI Bill debt for housing debt?
    <i class="fas fa-angle-down open"></i>
  </span>
</button>
```

## WCAG or Vendor Guidance (optional)

* [Ensure that links with the same accessible name serve a similar purpose](https://dequeuniversity.com/rules/axe/3.5/identical-links-same-purpose)

## Screenshots or Trace Logs

Each of the "Why might I have this debt?" links will not communicate the context for screen reader users, and need additional text to distinguish their different purposes.
![Screen Shot 2020-08-12 at 11 38 40 AM](https://user-images.githubusercontent.com/57469/90036720-586b0a00-dc91-11ea-9495-c2727092c6d4.png)

**Comment by bradl3yC** [comment]
We're no longer planning on releasing the MVP and we're going straight to V2 - This code will be removed during that process
